### PR TITLE
Resolve existing Rubocop offenses

### PR DESCRIPTION
**Pull request by samford**
This is just a bunch of simple fixes to resolve all the existing Rubocop offenses (and fix a syntax error in the `.rubocop.yml` file).

### CODE REVIEW

**Comment by jgeiger** [comment]
I would probably go all the way and make sure there are double quotes everywhere. It seems it's only checked in a few places.

**Comment by samford** [comment]
I ran `rubocop` in the root directory so it would check all the files and this PR resolves all those offenses.  I just went through the files by hand and I don't see anything that Rubocop missed.

The only double quotes that remain are in `connection.rb` and `connection_spec.rb` but there's string interpolation in those instances.  If you were to change those double quotes to single quotes then it would break the string interpolation, so Rubocop rightfully ignores those (i.e. it will only complain about the use of double quotes with no string interpolation).

Am I missing something else?

File: spec/newsblurry/story_spec.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@
       'story_authors' => 'The Big Picture',
       'story_hash' => '1095:67324d',
       'story_date' => '2013-10-07 23:54:16',
-      'story_tags' => ['Fukushima', 'Japan', 'tsunami'],
+      'story_tags' => %w('Fukushima', 'Japan', 'tsunami'),
       'story_content' => 'In 2011 a massive earthquake and tsunami wrecked ' \
                          'the Fukushima nuclear plant, resulting in a ' \
                          'meltdown that became the world\'s worst atomic ' \
```
Comment (jgeiger) [inline spec/newsblurry/story_spec.rb]: Double Quotes?

File: spec/newsblurry/connection_spec.rb [hunk 2]
```diff
@@ -7,13 +7,13 @@
...
-        headers: {
-          'Cookie' => '',
-          'User-Agent' => "Newsblurry::Connection - #{Newsblurry::VERSION}",
-        }
-      )
+        .with(
+          body: 'password=test&username=test',
+          headers: {
+            'Cookie' => '',
+            'User-Agent' => "Newsblurry::Connection - #{Newsblurry::VERSION}"
```
Comment (jgeiger) [inline spec/newsblurry/connection_spec.rb]: Double Quotes?

File: newsblurry.gemspec [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # -*- encoding: utf-8 -*-
-$:.push File.expand_path("../lib", __FILE__)
-require "newsblurry/version"
+$LOAD_PATH.push File.expand_path('../lib', __FILE__)
+require 'newsblurry/version'
```
Comment (jgeiger) [inline newsblurry.gemspec]: Double Quotes?

**Comment by jgeiger** [comment]
I added comments to a few places where there are still single quotes.

**Comment by samford** [comment]
Ah, I assumed you misspoke and meant to say "make sure there are single quotes everywhere".

The rule that Rubocop enforces is that you only want to use double quotes when you're doing string interpolation (i.e. `"here's some string interpolation #{variable_name}"`.  If you're not doing string interpolation then you should use single quotes (which don't handle string interpolation and are a little faster as a result).

**Comment by jgeiger** [comment]
Actually, I've moved to using double quotes everywhere. You can enforce that in rubocop.

Based on a number of posts I've read, (and personal benchmarking) there is no benefit to using single quotes. I was in the single quote camp before (as you can see in the code) but have switched sides after having done my own testing.

**Comment by samford** [comment]
Righto.  I added the associated rule to enforce your double quote preference in the `.rubocop.yml` file and converted all the single quotes to double quotes (resolving all the Rubocop offenses), so we should be good now.

**Pull request merged**